packages/xo-server/src/xo-mixins/metadata-backups.js

```diff
@@ -354,14 +354,20 @@ export default class metadataBackup {
               // 'readable-stream/pipeline' not call the callback when an error throws
               // from the readable stream
               stream.pipe(outputStream)
-              return fromEvent(stream, 'end').catch(error => {
-                if (error.message !== 'aborted') {
-                  throw error
-                }
-              })
+              return timeout.call(
+                fromEvent(stream, 'end').catch(error => {
+                  if (error.message !== 'aborted') {
+                    throw error
+                  }
+                }),
+                6e4
+              )
             })(),
             handler.outputFile(metaDataFileName, metadata),
-          ])
+          ]).catch(error => {
+            handler.rmtree(dir)::ignoreErrors()
```

Comment:
What if the first promise rejects and we remove the tree before the second one?

Reply:
Right :+1: , we should use an `asyncMap` instead.